planning/route_handler/src/route_handler.cpp

```diff
@@ -1319,6 +1319,61 @@ int RouteHandler::getNumLaneToPreferredLane(
   return 0;  // TODO(Horibe) check if return 0 is appropriate.
 }
 
+double RouteHandler::getLateralDistanceToPreferredLane(
+  const lanelet::ConstLanelet & lanelet, const Direction direction) const
+{
+  if (exists(preferred_lanelets_, lanelet)) {
+    return 0.0;
+  }
+
+  double accumulated_distance = 0.0;
+  lanelet::ConstLanelet current_lanelet = lanelet;
+
+  if ((direction == Direction::NONE) || (direction == Direction::RIGHT)) {
+    const auto & right_lanes =
+      lanelet::utils::query::getAllNeighborsRight(routing_graph_ptr_, lanelet);
+    for (const auto & right : right_lanes) {
+      const auto & current_centerline = current_lanelet.centerline();
+      const auto & next_centerline = right.centerline();
+      if (current_centerline.empty() || next_centerline.empty()) {
+        return -accumulated_distance;
+      }
+      const auto & curr_pt = current_centerline[0];
+      const auto & next_pt = next_centerline[0];
+      accumulated_distance += lanelet::geometry::distance2d(to2D(curr_pt), to2D(next_pt));
+
+      if (exists(preferred_lanelets_, right)) {
+        return -accumulated_distance;
+      }
+      current_lanelet = right;
+    }
+  }
+
+  accumulated_distance = 0.0;
+  current_lanelet = lanelet;
```

Comment:
Same with previous suggestion, it might be nicer to reinitialize then in the if loop.

Happy to hear about your thoughts on this matter.

Reply:
I agree with this one as well. 

Fixed https://github.com/autowarefoundation/autoware.universe/pull/3361/commits/ad4311bc2e6c46826c462f66cbe43197901b1c93

Reply:
@zulfaqar-azmi-t4